### PR TITLE
feat(create-waku): bundle into single file

### PR DESCRIPTION
**Pull request by himself65**

### CODE REVIEW

File: packages/create-waku/tsconfig.json [hunk 1]
```diff
@@ -1,12 +1,12 @@
 {
   "extends": "../../tsconfig.json",
   "compilerOptions": {
-    "rootDir": "./src",
+    "rootDir": ".",
     "outDir": "./dist",
```
Comment (dai-shi) [inline packages/create-waku/tsconfig.json:5]: Is it a convention or a standard?
I'm not super comfortable with root being `.` and out being `./dist`.

**Comment by himself65** [comment]
I have to rename the file into `index.ts`, cause the output entry will always be `index.js`.

File: packages/create-waku/src/index.ts [hunk 2]
```diff
@@ -1,12 +1,11 @@
 #!/usr/bin/env node
-
 import { existsSync, readdirSync } from 'node:fs';
 import fsPromises from 'node:fs/promises';
-import path from 'node:path';
-import { fileURLToPath } from 'node:url';
+import path, { dirname } from 'node:path';
```
Comment (dai-shi) [inline packages/create-waku/src/index.ts:4]: Not a big fan of mixing default and named. Can you go with all named?

File: packages/create-waku/src/index.ts [hunk 3]
```diff
@@ -1,12 +1,11 @@
...
 import { default as prompts } from 'prompts';
 import { red, green, bold } from 'kolorist';
 import fse from 'fs-extra/esm';
+import { fileURLToPath } from 'node:url';
```
Comment (dai-shi) [inline packages/create-waku/src/index.ts:8]: move up to the original position.

File: packages/create-waku/src/index.ts [hunk 4]
```diff
@@ -31,9 +30,12 @@ function canSafelyOverwrite(dir: string) {
 async function init() {
   let targetDir = '';
   const defaultProjectName = 'waku-project';
-
-  const templateRoot = fileURLToPath(new URL('../template', import.meta.url));
-  const CHOICES = await fsPromises.readdir(templateRoot);
+  const __dirname = dirname(fileURLToPath(import.meta.url));
+  const templateRoot = path.resolve(__dirname, '../template');
```
Comment (dai-shi) [inline packages/create-waku/src/index.ts:34]: Can we simplify like this?
```suggestion
  const templateRoot = path.join(fileURLToPath(import.meta.url), '../../template');
```

**Review by dai-shi — COMMENTED** [review]
Looks good. Some refactor ideas.

**Review by dai-shi — APPROVED** [review]
Not familiar with https://github.com/vercel/ncc, but looks okay.

**Pull request merged**